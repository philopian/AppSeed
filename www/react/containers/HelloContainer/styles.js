import styled from "styled-components";
import colors from "../../constants/colors";

// Overriding styles
export const Wrapper = styled.div`
  color: ${colors.white};
  background-color: ${colors.blue};
  padding: 10px;

  .from-store {
    font-weight: 700;
  }
`;
