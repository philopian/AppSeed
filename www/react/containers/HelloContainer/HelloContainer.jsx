import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Wrapper } from "./styles.js";

import HelloComponent from "../../components/HelloComponent";
import { fetchSampleData } from "../../actions/hello.js";

export class HelloContainer extends Component {
  state = {
    yy: ""
  };

  componentWillMount() {
    this.props.fetchSampleData();
  }

  render() {
    return (
      <Wrapper>
        <h1>Hello containter!</h1>
        <p>Container components connect to the redux store for props</p>
        <p>
          this is from the redux store --->{" "}
          <span className="from-store">{this.props.hello}</span>
        </p>
        <p>
          this is from the redux store via REST --->{" "}
          <span className="from-store">it {this.props.sampleData.yes}</span>
        </p>
        <HelloComponent message={this.props.sampleData.yes} />
      </Wrapper>
    );
  }
}

HelloContainer.propTypes = {
  hello: PropTypes.string,
  sampleData: PropTypes.object,
  fetchSampleData: PropTypes.func
};
const mapStateToProps = state => ({
  hello: state.hello.message,
  sampleData: state.hello.sampleData
});
const mapDispatchToProps = dispatch => ({
  fetchSampleData: () => dispatch(fetchSampleData())
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HelloContainer);
