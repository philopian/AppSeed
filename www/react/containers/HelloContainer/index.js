import HelloContainer from "./HelloContainer.jsx";
export default HelloContainer;
