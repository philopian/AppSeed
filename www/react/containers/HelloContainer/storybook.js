import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { setConsoleOptions } from "@storybook/addon-console";

import React from "react";
import { HelloContainer as Component } from "./HelloContainer.jsx";

setConsoleOptions({
  panelExclude: [/[HMR]/]
});

const Info = {
  componentSection: "Container",
  title: "HelloContainer",
  about: "this is a simple example container",
  mockProps: {
    hello: "welcome user",
    sampleData: { yes: "a little sample data for you" },
    fetchSampleData: () => action("fetchSampleData")
  }
};

storiesOf(Info.componentSection, module)
  .addDecorator(withKnobs)
  .add(
    Info.title,
    withInfo(Info.about)(() => <Component {...Info.mockProps} />)
  );
