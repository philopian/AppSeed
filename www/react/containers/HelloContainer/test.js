import React from "react";
import Enzyme, { configure, shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import { HelloContainer as Component } from "./HelloContainer.jsx";

Enzyme.configure({ adapter: new Adapter() });

const mockLoginfn = jest.fn();
const props = {
  hello: "......",
  sampleData: {},
  fetchSampleData: mockLoginfn
};

describe("HelloContainer (Snapshot)", () => {
  it("HelloContainer renders without crashing", () => {
    const mockLoginfn = jest.fn();
    const component = renderer.create(<Component {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe("Component", () => {
  let component;
  const mockLoginfn = jest.fn();
  beforeEach(() => {
    component = shallow(<Component {...props} />);
  });

  // ...tests here...
});

// Logic
describe("Addition", () => {
  it("knows that 2 and 2 make 4", () => {
    expect(2 + 2).toBe(4);
  });
});
