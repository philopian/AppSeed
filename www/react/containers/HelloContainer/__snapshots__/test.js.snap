// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloContainer (Snapshot) HelloContainer renders without crashing 1`] = `
<div
  className="sc-bdVaJa bixvQK"
>
  <h1>
    Hello containter!
  </h1>
  <p>
    Container components connect to the redux store for props
  </p>
  <p>
    this is from the redux store ---&gt;
     
    <span
      className="from-store"
    >
      ......
    </span>
  </p>
  <p>
    this is from the redux store via REST ---&gt;
     
    <span
      className="from-store"
    >
      it 
    </span>
  </p>
  <div
    className="my-component"
  >
    <div
      className="sc-bwzfXH bPsJDo"
    >
      <h1>
        HelloComponent
      </h1>
      <p>
        it 
        <span>
          World
        </span>
      </p>
    </div>
  </div>
</div>
`;
